tests/pytests/test_hybrid_timeout_threadpool.py

```diff
@@ -0,0 +1,149 @@
+               'PARAMS', '2', 'BLOB', query_vector, 'TIMEOUT_AFTER_N_SEARCH', '1', 
+               'DEBUG_PARAMS_COUNT', '2').error().contains('Timeout')
+
+@skip(cluster=True)
+def test_depleter_timeout_return():
+    """Test RPDepleter timeout detection with RETURN policy"""
+    env = Env(enableDebugCommand=True, moduleArgs='ON_TIMEOUT RETURN')
+    setup_basic_index(env)
+    
+    # Use debug parameters to simulate timeout
+    response = env.cmd('_FT.DEBUG', 'FT.HYBRID', 'idx', 'SEARCH', 'running', 'VSIM', '@embedding', '$BLOB', 
+                       'PARAMS', '2', 'BLOB', query_vector, 'TIMEOUT_AFTER_N_SEARCH', '1', 
+                       'DEBUG_PARAMS_COUNT', '2')
+    
+    # Should return partial results with timeout warning
+    warnings = get_warnings(response)
+    env.assertTrue(len(warnings) > 0, "Expected timeout warning")
+    env.assertTrue(any('SEARCH' in warning for warning in warnings), "Expected SEARCH timeout in warnings")
+
+# Test HybridMerger timeout handling
+@skip(cluster=True)
+def test_hybrid_merger_timeout_fail():
+    """Test HybridMerger timeout handling with FAIL policy"""
+    env = Env(enableDebugCommand=True, moduleArgs='ON_TIMEOUT FAIL')
+    setup_basic_index(env)
+    
+    # Use debug parameters to simulate timeout in tail pipeline
+    env.expect('_FT.DEBUG', 'FT.HYBRID', 'idx', 'SEARCH', 'running', 'VSIM', '@embedding', '$BLOB', 
+               'PARAMS', '2', 'BLOB', query_vector, 'TIMEOUT_AFTER_N_TAIL', '1', 
+               'DEBUG_PARAMS_COUNT', '2').error().contains('Timeout')
+
+@skip(cluster=True)
+def test_hybrid_merger_timeout_return():
+    """Test HybridMerger timeout handling with RETURN policy"""
+    env = Env(enableDebugCommand=True, moduleArgs='ON_TIMEOUT RETURN')
+    setup_basic_index(env)
+    
+    # Use debug parameters to simulate timeout in tail pipeline
+    response = env.cmd('_FT.DEBUG', 'FT.HYBRID', 'idx', 'SEARCH', 'running', 'VSIM', '@embedding', '$BLOB', 
+                       'PARAMS', '2', 'BLOB', query_vector, 'TIMEOUT_AFTER_N_TAIL', '1', 
+                       'DEBUG_PARAMS_COUNT', '2')
+    
+    # Should return partial results with timeout warning
+    warnings = get_warnings(response)
+    env.assertTrue(len(warnings) > 0, "Expected timeout warning")
+
+# Test timeout policy consistency
+@skip(cluster=True)
+def test_timeout_policy_consistency():
+    """Test that timeout policies work consistently across all components"""
+    # Test FAIL policy
+    env_fail = Env(enableDebugCommand=True, moduleArgs='ON_TIMEOUT FAIL')
+    setup_basic_index(env_fail)
+    
+    env_fail.expect('FT.HYBRID', 'idx', 'SEARCH', 'running', 'VSIM', '@embedding', '$BLOB', 
+                    'PARAMS', '2', 'BLOB', query_vector, 'TIMEOUT', '1').error().contains('Timeout')
+    
+    # Test RETURN policy
+    env_return = Env(enableDebugCommand=True, moduleArgs='ON_TIMEOUT RETURN')
```

Comment:
Why do we need a new env? Can't we reuse the one created previously and only config the timeout policy?